ledger/src/blockstore.rs

```diff
@@ -11031,7 +11017,7 @@ pub mod tests {
                                 blockstore.lowest_cleanup_slot.write().unwrap();
                             blockstore.insert_shreds(shreds, None, false).unwrap();
                             assert!(blockstore.get_duplicate_slot(slot).is_some());
-                            assert!(blockstore.is_dead(slot));
+                            assert!(!blockstore.is_dead(slot));
```

Comment:
Now since we are including well-formed FEC sets the slot will be duplicate but not dead.

Reply:
Yeah in the original setup it was marking 2 shreds in the same FEC set as last_in_slot which resulted in it getting marked dead.

Reply:
Hmm, this test is VERY slot on my machine; were you able to run it quickly on your box ?
```
test blockstore::tests::test_get_slot_entries_dead_slot_race has been running for over 60 seconds
```
Note that the tip of master returned in < 1 second for me:
```
$ cargo test test_get_slot_entries_dead_slot_race
...
running 1 test
test blockstore::tests::test_get_slot_entries_dead_slot_race ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 370 filtered out; finished in 0.31s
```

Reply:
Addressed in https://github.com/anza-xyz/agave/pull/6163/commits/ecaf48ed56d5bd4c812f5416fe8e3f8a560bdf69. I've also switched this to scoped threads so it will actually crash rather than hanging when any asserts fail.